server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceComputer.java

```diff
@@ -293,6 +310,11 @@ public DesiredBalance compute(
                 for (final var shardRouting : routingNode) {
                     if (shardRouting.initializing()) {
                         hasChanges = true;
+                        if (shardRouting.primary()
```

Comment:
For now this only considers primaries. Do we need to consider search replicas for serverless (at least one search replica as we did in https://github.com/elastic/elasticsearch/pull/113847)? They could have the same issue, I think.

Reply:
Yeah, sounds reasonable. It would be a good idea to include that.